### PR TITLE
Automate build process (WIP)

### PR DESCRIPTION
**Pull request by Lundis**
Not much progress yet. Created the basic dockerfile which should have everything needed to compile. Completely untested so far.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+RUN wget -O /opt/android-ndk.zip https://dl.google.com/android/repository/android-ndk-${ANDROID_NDK_VERSION}-linux.zip; \
+    unzip -q /opt/android-ndk.zip -d /opt; \
+    rm /opt/android-ndk.zip
+
+FROM android_ndk as macos
+
+WORKDIR /opt
+
+RUN git clone https://github.com/tpoechtrager/osxcross; \
+    wget -P osxcross/tarballs https://go-sdl2.veandco.sg-sin1.upcloudobjects.com/MacOSX10.14.sdk.tar.xz; \
```
Comment (veeableful) [inline docker/Dockerfile]: Just an FYI, the SDKs have been moved to the following URLs!
```
https://s3.veand.co/go-sdl2/MacOSX10.14.sdk.tar.xz
https://s3.veand.co/go-sdl2/MacOSX11.3.sdk.tar.xz
```

Comment (Lundis) [reply docker/Dockerfile]: Thanks. Hopefully I'll finally have time to finish this now during my vacation... :D

**Comment by Lundis** [comment]
Getting closer to working... Just need to add some cross-compilation flags to cmake/configure for android and darwin

```
configure:3086: ./conftest
../configure: line 3088: ./conftest: cannot execute binary file: Exec format error
configure:3090: $? = 126
configure:3097: error: in `/buildscripts/SDL2_gfx-1.0.4/build-android_arm':
configure:3099: error: cannot run C compiled programs.
If you meant to cross compile, use `--host'.
```

**Comment by veeableful** [comment]
Thanks @Lundis. I have also experimented with Github Actions to generate the libraries as artifacts and using pre-built Docker image would help speed it up a lot!

**Comment by Lundis** [comment]
I'm closing this. It doesn't really make sense to spend time maintaining a docker image for this, seeing how rarely it needs to be ran. I also didn't know anything about github actions/runners when I started this.

**Pull request closed without merging**